services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,37 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.{
+  ProjectConfigurationRequest,
+  ProjectConfigurationResponse
+}
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationRequest): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .replaceOne(
+        toDocument(ProjectConfiguration(configuration.projectId, configuration.projectFileConfigurations)),
+        "projectId",
+        configuration.projectId.value
+      )
       .map(_.toString)
 
-  def getById(projectId: ProjectId): Future[Option[ProjectConfiguration]] =
-    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument))
+  def getById(projectId: ProjectId): Future[Option[ProjectConfigurationResponse]] =
+    repository.getByParam("projectId", projectId.value).map(_.headOption.map(fromDocument)).map {
+      case Some(conf) if conf.isActive =>
+        Some(ProjectConfigurationResponse(conf.projectId, conf.projectFileConfigurations))
+      case _ => None
+    }
+
+  def deactivateConfiguration(projectId: ProjectId): Future[String] =
```

Comment:
I'm not sure, is it a good thing that we have output result as a `String`? What other options do we have?

Reply:
Repository returns ```Future[UpdateResult]```, is it better than ```String```?